### PR TITLE
Improve the guide about merging the tmp branch into main

### PR DESCRIPTION
**Pull request by Kerollmops**
This PR improves the part of the guide when it comes to fixing merge conflicts and when we need to create a temporary branch.

### CODE REVIEW

File: resources/meilisearch-release.md [hunk 1]
```diff
@@ -98,9 +98,14 @@ Some commits might miss to `main` since the engine-team did some hotfixes merged
 In this case, you need to bring them back from `release-vX.Y.Z` to `main` by merging a PR originating `release-vX.Y.Z` and pointing to `main`.
```
Comment (curquiza) [inline resources/meilisearch-release.md:98]:
```suggestion
In this case, you need to bring them back from `release-vX.Y.Z` to `main` by creating and merging a PR originating `release-vX.Y.Z` and pointing to `main`.
```

File: resources/meilisearch-release.md [hunk 2]
```diff
@@ -98,9 +98,14 @@ Some commits might miss to `main` since the engine-team did some hotfixes merged
...
 
 ⚠️ If you encounter any git conflicts when creating the PR, please do NOT fix the git conflicts directly on the `release-vX.Y.Z` branch. It would bring the changes present in `main` into `release-vX.Y.Z`, which would break a potential future patched release.
+
+![GitHub interface showing merge conflicts](../assets/merge-conflicts.png)
+
 Instead:
 - Create a new branch originating `release-vX.Y.Z`, like `tmp-release-vX.Y.Z`
 - Fix the git conflicts on this new branch
```
Comment (curquiza) [inline resources/meilisearch-release.md:106]:
```suggestion
- Create a PR from the `tmp-release-vX.Y.Z` branch and pointing to `main`
- Fix the git conflicts on this new branch
```

File: resources/meilisearch-release.md [hunk 3]
```diff
@@ -98,9 +98,14 @@ Some commits might miss to `main` since the engine-team did some hotfixes merged
...
+    - By either creating a Pull Request and merging throught the GitHub interface
```
Comment (curquiza) [inline resources/meilisearch-release.md:107]:
```suggestion
    - Fixing the git conflict via the GitHub interface
```

Comment (curquiza) [reply resources/meilisearch-release.md:107]: You must create a PR anyway

File: resources/meilisearch-release.md [hunk 4]
```diff
@@ -98,9 +98,14 @@ Some commits might miss to `main` since the engine-team did some hotfixes merged
...
+    - By pulling the `main` branch into `temp-release-vX.Y.Z`
 - Merge this new branch into `main`
```
Comment (curquiza) [inline resources/meilisearch-release.md:109]:
```suggestion
- Merge this new PR into `main`
```

**Comment by curquiza** [comment]
@Kerollmops I modified a little bit your suggestion: when talking with you, your main questioning was about what to do (should I use and create a PR or not), and your changed did not really reflect this

File: resources/meilisearch-release.md [hunk 5]
```diff
@@ -98,9 +98,14 @@ Some commits might miss to `main` since the engine-team did some hotfixes merged
...
 
 ⚠️ If you encounter any git conflicts when creating the PR, please do NOT fix the git conflicts directly on the `release-vX.Y.Z` branch. It would bring the changes present in `main` into `release-vX.Y.Z`, which would break a potential future patched release.
+
+![GitHub interface showing merge conflicts](../assets/merge-conflicts.png)
+
 Instead:
 - Create a new branch originating `release-vX.Y.Z`, like `tmp-release-vX.Y.Z`
 - Fix the git conflicts on this new branch
+    - By either creating a Pull Request and merging throught the GitHub interface
+    - By pulling the `main` branch into `temp-release-vX.Y.Z`
```
Comment (curquiza) [inline resources/meilisearch-release.md:108]:
```suggestion
    - By pulling the `main` branch into `temp-release-vX.Y.Z` and fixing them on your machine.
```

**Comment by Kerollmops** [comment]
Yup, thank you for your review.

> your main questioning was about what to do (should I use and create a PR or not), and your changes did not really reflect this.

I just noticed that when you create a pull request and there are merge conflicts there is a basic explanation of how to fix them on your side in a tab at the bottom of the PR.

**Review by curquiza — APPROVED** [review]

**Pull request merged**